__tests__/app.test.js

```diff
@@ -0,0 +1,38 @@
+const { app } = require('../app')
+const seed = require('../db/seeds/seed')
+const db = require('../db/connection')
+const data = require('../db/data/test-data')
+const request = require('supertest')
+
+beforeEach(() => {
+    return seed(data);
+})
+
+afterAll(() => {
+    return db.end();
+})
+
+describe('app', () => {
+    describe('GET /api/categories', () => {
+        test('Responds with a 200 status code', () => {
+            return request(app).get('/api/categories').expect(200)
+        });
+        test('Responds with an array', () => {
+            return request(app).get('/api/categories').expect(200).then((response) => {
+                expect(Array.isArray(response.body)).toBe(true)
+            });
+        });
+        test('Responds with an array', () => {
+            return request(app).get('/api/categories').expect(200).then((response) => {
+                const categories = response.body
+                categories.forEach((category) => {
+                    expect(category).toHaveProperty('slug')
+                    expect(category).toHaveProperty('description')
+                })
+            });
+        });
+        test('Responds with an error 404 incorrect endpoints', () => {
```

Comment:
Excellent having this test 404 but it could be extracted out from this particular endpoint describe block so it is a test for the app as a whole, (hint app.all() and a 404 error handler function that sends a message too) 😄